### DISCUSSION TITLE
Modifying the fieldset data

### DISCUSSION THREAD

**Discussion by philippemiron** [Parcels-code/Parcels#1032] [Q&A] [answered]
Hi everyone,

I have velocity fields with some erroneous values that I would like to identify and modify before executing the kernel. For example, I would like to remove the values higher than 10m/s but this does not work with the *lazy* loading going on. 

```
fieldset.U.data[fieldset.U.data > 10] = 0
TypeError: '>' not supported between instances of 'DeferredArray' and 'int'
```

Someone could point me to an example on how to proceed?

Thanks ! 🙏

**Comment by erikvansebille** [comment, marked as answer]
You can use `vmax` (and `vmin`) when you create a FieldSet to replace any values higher than a certain maximum by 0. See also https://github.com/OceanParcels/parcels/blob/master/parcels/field.py#L168-L169
